model/flow/filter/identity.go

```diff
@@ -7,14 +7,23 @@ import (
 	"github.com/onflow/flow-go/model/flow"
 )
 
+// Adapt adapts a filter for a specific identity type.
+//
+//	Converts flow.IdentityFilter[flow.IdentitySkeleton] to flow.IdentityFilter[flow.Identity].
+func Adapt(f flow.IdentityFilter[flow.IdentitySkeleton]) flow.IdentityFilter[flow.Identity] {
```

Comment:
I think that is a good reasoning. 

Overall, I am wondering if we should maybe remove the [`In`](https://github.com/onflow/flow-go/blob/739cbdf96f6ada2296ecf794a2f53340f92118d6/model/flow/filter/identity.go#L55-L59) and [`Adapt`](https://github.com/onflow/flow-go/blob/739cbdf96f6ada2296ecf794a2f53340f92118d6/model/flow/filter/identity.go#L10-L17) method all together. Reading your comment, I realized that my issue with the functions is not the ambiguous name, but rather the incomplete documentation. Because, _neither_ goDoc states that we are in fact only filtering by NodeID, ignoring all other properties. For example, assume we created the filter including identity X with `weight=10` and later we are calling the filter with X' with `weight=8`. Is the filter supposed to return true, despite the weight being different? 

I feel replacing  [`In`](https://github.com/onflow/flow-go/blob/739cbdf96f6ada2296ecf794a2f53340f92118d6/model/flow/filter/identity.go#L55-L59) and [`Adapt`](https://github.com/onflow/flow-go/blob/739cbdf96f6ada2296ecf794a2f53340f92118d6/model/flow/filter/identity.go#L10-L17) by
```golang
HasNodeID[T](list.NodeIDs()...)
```
would avoid this ambiguity all together. In the entire code base, we have 8 usages of `In` and `Adapt` combined, so it wouldn't be a big refactoring altogether.

Reply:
Yup, `Adapt` is a general purpose function to adapt filters from smaller subset to larger one. Why adapt? Because it adapts interfaces.